src/pybind/mgr/dashboard/frontend/src/app/shared/directives/disable-edit.directive.ts

```diff
@@ -0,0 +1,27 @@
+import { Directive, HostBinding, Input } from '@angular/core';
+import { Permissions } from '../models/permissions';
+import { AuthStorageService } from '../services/auth-storage.service';
+
+@Directive({
+  selector: '[cdDisableEdit]'
+})
+export class DisableEditDirective {
```

Comment:
How about `updateOnlyDirective` as `disableEditDirective` sounds misleading.
IMO `cdUpdateOnly="osd"` sounds better than `cdDisableEdit="osd"`. May be there is also a better way to name it. May be something like `usableWithUpdatePrem`?

As this is used on multiple locations the name should really describe why this directive is used here without having to look it up.

Reply:
I agree. I am also not finalised with the name of the directive. Its like 'Open for suggestions'. Thank you for your suggestion :)